### PR TITLE
Fix val_check_interval with fast_dev_run

### LINKED ISSUE #5538: fast_dev_run breaks with val_check_interval
## 🐛 Bug

It looks like `val_check_interval` must be <= `num_training_batches`, but the latter is set to 1 in `fast_dev_run`, so things break.

## Please reproduce using the BoringModel

https://colab.research.google.com/drive/1cMWbhk5pmkGe_y6znQFHebRUrqa_ssu9?usp=sharing
### To Reproduce

Use following [**BoringModel**](https://colab.research.google.com/drive/1HvWVVTK8j2Nj52qU4Q4YCyzOm0_aLQF3?usp=sharing) and post here

### Expected behavior

Probably validation checks should be disabled in `fast_dev_run` Or `val_check_interval` should be set to 1 as well.

### Environment

* CUDA:
	- GPU:
		- Tesla T4
	- available:         True
	- version:           10.1
* Packages:
	- numpy:             1.19.5
	- pyTorch_debug:     True
	- pyTorch_version:   1.7.0+cu101
	- pytorch-lightning: 1.1.4
	- tqdm:              4.41.1
* System:
	- OS:                Linux
	- architecture:
		- 64bit
		- 
	- processor:         x86_64
	- python:            3.6.9
	- version:           #1 SMP Thu Jul 23 08:00:38 PDT 2020

### PR DESCRIPTION
**Pull request by rohitgr7**
## What does this PR do?

Fixes #5538

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified
 - [x] **Check that target branch and milestone match!**

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: tests/trainer/flags/test_fast_dev_run.py [hunk 1]
```diff
@@ -36,30 +36,59 @@ def test_callbacks_and_logger_not_called_with_fastdevrun(tmpdir, fast_dev_run):
     class FastDevRunModel(BoringModel):
```
Comment (awaelchli) [inline tests/trainer/flags/test_fast_dev_run.py:36]: minor suggestion, you can do 
`model = Mock(wraps=FastDevRunModel())` and then
anywhere 
`model.training_step.call_count  == ... `
which would remove a few lines of boilerplate code from this test :)

**Review by awaelchli — APPROVED** [review]

File: tests/trainer/flags/test_fast_dev_run.py (hunk 1 above)
Comment (rohitgr7) [reply tests/trainer/flags/test_fast_dev_run.py:36]:
```python
../pytorch_lightning/utilities/model_utils.py:42: in is_overridden    
is_overridden = instance_attr.__code__ is not super_attr.__code__ 
```
```python
def __getattr__(self, name):                                              
    if name in {'_mock_methods', '_mock_unsafe'}:                         
        raise AttributeError(name)                                        
    elif self._mock_methods is not None:                                  
        if name not in self._mock_methods or name in _all_magics:         
            raise AttributeError("Mock object has no attribute %r" % name)
    elif _is_magic(name):
        raise AttributeError(name)
        AttributeError: __code__
```
getting some error with this. any idea how to resolve?

Comment (Borda) [reply tests/trainer/flags/test_fast_dev_run.py:36]: Let's land it now and prepare a refactor on all these call counts...
@awaelchli mind open an issue for this refactor and ping me there :]

**Review by Borda — APPROVED** [review]

File: tests/trainer/flags/test_fast_dev_run.py (hunk 1 above)
Comment (awaelchli) [reply tests/trainer/flags/test_fast_dev_run.py:36]: yes it's ok. 
@rohitgr7 sorry thought it would be straightforward but ok, we can look at it later :)

**Review by justusschock — APPROVED** [review]

**Review by tchaton — APPROVED** [review]

**Pull request merged**